### PR TITLE
feat: join workers

### PR DESCRIPTION
**Pull request by Mishavint**
closes issue #10

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -9,7 +9,8 @@ k8s_cluster_apt_repository: 'deb [signed-by=/usr/share/keyrings/kubernetes-archi
 
 k8s_cluster_kubernetes_version: 1.26.0
 k8s_cluster_kubelet_config_root_dir: '/etc/kubernetes'
-k8s_cluster_ignore_preflight_errors: "all"
+k8s_cluster_init_ignore_preflight_errors: "all"
+k8s_cluster_join_ignore_preflight_errors: "all"
```
Comment (bissquit) [inline defaults/main.yml:13]: Try to move those settings into kubeadm config

File: molecule/master_with_worker/converge.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  become: yes
+  become_method: su
+  gather_facts: false
+  pre_tasks:
+    - name: Install prerequisites
+      ansible.builtin.apt:
+        name:
+          - iproute2  # provides network facts
+          - kmod  # modprobe and lsmod
+          - net-tools
```
Comment (bissquit) [inline molecule/master_with_worker/converge.yml:35]: Add curl installation into all test scenarios or at least check if curl is already installed

File: molecule/verify-common.yml [hunk 3]
```diff
@@ -33,18 +33,6 @@
...
-    - name: Debug
-      ansible.builtin.debug:
-        var: services_state.ansible_facts.services['kubelet.service']
-
-    - name: Check if etcd service is in running state
-      ansible.builtin.assert:
-        that: services_state.ansible_facts.services['kubelet.service'].state == "running"
-      ignore_errors: yes
-      register: etcd_state
+- name: Include kubelet
```
Comment (bissquit) [inline molecule/verify-common.yml:36]: Move to top-level please

File: molecule/verify-workers.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    - name: Debug kubectl_show for {{ ansible_hostname }}
+      ansible.builtin.debug:
+        var: kubectl_show
+
+- name: Check that workers nodes are ready
+  delegate_to: "{{ groups['control_plane'][0] }}"
+  block:
+    - name: Check for node to be ready
+      ansible.builtin.pause:
+        minutes: 3
```
Comment (bissquit) [inline molecule/verify-workers.yml:21]: Think about changing to the following scheme: checking `http://localhost:10248/healthz` endpoint until 200 status

Note: check please what this url actually shows during worker node initialization

**Review by bissquit — CHANGES_REQUESTED** [review]

**Review by bissquit — CHANGES_REQUESTED** [review]

File: defaults/main.yml [hunk 5]
```diff
@@ -9,7 +9,8 @@ k8s_cluster_apt_repository: 'deb [signed-by=/usr/share/keyrings/kubernetes-archi
 
 k8s_cluster_kubernetes_version: 1.26.0
 k8s_cluster_kubelet_config_root_dir: '/etc/kubernetes'
-k8s_cluster_ignore_preflight_errors: "all"
+k8s_cluster_init_extra_opts: --ignore-preflight-errors-all
+k8s_cluster_join_command_extra_opts: --ignore-preflight-errors-all
```
Comment (bissquit) [inline defaults/main.yml]: Return back previous variable please. We could change it on host- or group- level of Ansible variables

**Review by bissquit — CHANGES_REQUESTED** [review]

File: defaults/main.yml (hunk 5 above)
Comment (bissquit) [inline defaults/main.yml]: Is that a typo?

```k8s_cluster_init_extra_opts: --ignore-preflight-errors-all```

Maybe right case should be the following:

```k8s_cluster_init_extra_opts: "--ignore-preflight-errors=all"```

**Review by bissquit — CHANGES_REQUESTED** [review]

File: molecule/default_with-worker/molecule.yml [hunk 6]
```diff
@@ -16,6 +16,7 @@ platforms:
       - /var/lib/containerd
     privileged: true
     pre_build_image: true
+    keep_volumes: false
```
Comment (bissquit) [inline molecule/default_with-worker/molecule.yml:19]: What this option actually does?

**Review by bissquit — CHANGES_REQUESTED** [review]

File: tasks/control-plane-init.yml [hunk 7]
```diff
@@ -61,14 +61,13 @@
     k8s_cluster_join_command: "{{ kubernetes_join_command_result.stdout | default('') }}"
   delegate_to: "{{ item }}"
   delegate_facts: true
-  with_items: "{{ groups['control_plane'] + groups['workers'] }}"
+  loop: "{{ groups['control_plane'] + groups['workers'] }}"
```
Comment (bissquit) [inline tasks/control-plane-init.yml:64]: Is there a way to avoid this solution? Ansible inventory groups hardcoding is a bad

**Review by bissquit — CHANGES_REQUESTED** [review]

File: molecule/default_with-worker/molecule.yml (hunk 6 above)
Comment (Mishavint) [reply molecule/default_with-worker/molecule.yml:19]: By default docker plugin saves volumes after destroying:

[source](https://github.com/ansible-community/molecule-plugins/blob/7bb0b345f00a473a78d05c350acbf10f4e928957/src/molecule_plugins/docker/playbooks/destroy.yml#LL25C9-L25C21)

File: tasks/control-plane-init.yml (hunk 7 above)
Comment (Mishavint) [reply tasks/control-plane-init.yml]: Could be be variables for groups, like [there](https://github.com/cloud-labs-infra/ansible-etcd-cluster/blob/ec990f914667bd94292901bd2f1a38d7c123f108/defaults/main.yml#LL8C1-L8C29)

**Review by bissquit — CHANGES_REQUESTED** [review]
Rename please `control-plane-init.yml` to `init-control-plane.yml`, and for workers the same

**Comment by bissquit** [comment]
Also try to bind `when` statement to some variables like `k8s_cluster_node_type: master` but not to inventory group names.

The same behavior for initial master node (but not forget to add preflight test to check if var e.g. `k8s_cluster_initial_master: true` belongs for the only host in play)

**Review by bissquit — APPROVED** [review]

**Pull request merged**